android/src/main/kotlin/com/example/another_location_plugin/AnotherLocationPlugin.kt

```diff
@@ -1,30 +1,137 @@
+            requestPermissions -> requestPermissions()
+            else -> result.notImplemented()
+        }
+    }
+
+    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
+    }
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activity = binding.activity
+        binding.addActivityResultListener(this)
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {
+        Log.i("onDetachFromAct(config)", "Not implemented")
+    }
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        Log.i("onReattachedToActivity", "Not implemented")
     }
-  }
 
-  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
-    channel.setMethodCallHandler(null)
-  }
+    override fun onDetachedFromActivity() {
+        Log.i("onDetachedFromActivity", "Not implemented")
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
+        Log.i("onActivityResult", "Not implemented")
+        return true
+    }
+
+    private fun getPlatformVersion(result: Result) {
+        result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    }
+
+    private fun checkPermissions(): Boolean {
+        var allowed = false
+        context?.run {
+            allowed = ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACCESS_COARSE_LOCATION
+            ) == PackageManager.PERMISSION_GRANTED
+            permissionEventsSource?.success(
+                if (allowed) {
+                    permissionAllowed
+                } else {
+                    permissionNotAllowed
+                }
+            )
+        }
+        return allowed
+    }
+
+    private fun requestPermissions() {
+        context?.run {
+            activity?.apply {
+                requestPermissions(
+                    this,
+                    arrayOf(
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    ),
+                    requestCode
+                )
+            }
+        }
+    }
+
+    companion object Constants {
+        const val pluginChannelName = "another_location_plugin"
+        const val permissionChannelName = "permission_event_channel"
+        const val getPlatformVersion = "getPlatformVersion"
+        const val checkPermissions = "checkPermissions"
+        const val requestPermissions = "requestPermissions"
+        const val permissionAllowed = "Permission allowed"
+        const val permissionNotAllowed = "Permission not allowed"
+        const val requestCode = 500
```

Comment:
the name mangling for the companion objects uses  MAYUS and the words separated by lower hyphen: 
 e.g. PLUGIN_CHANNEL_NAME

Reply:
Understand, done!